src/test/librados/TestCase.cc

```diff
@@ -283,17 +284,17 @@ void RadosTest::cleanup_namespace(rados_ioctx_t ioctx, std::string ns)
 {
   rados_ioctx_snap_set_read(ioctx, LIBRADOS_SNAP_HEAD);
   rados_ioctx_set_namespace(ioctx, ns.c_str());
-  rados_list_ctx_t list_ctx;
-  ASSERT_EQ(0, rados_nobjects_list_open(ioctx, &list_ctx));
+  std::unique_ptr<rados_list_ctx_t> list_ctx;
+  ASSERT_EQ(0, rados_nobjects_list_open(ioctx, list_ctx.get()));
   int r;
   const char *entry = NULL;
   const char *key = NULL;
-  while ((r = rados_nobjects_list_next(list_ctx, &entry, &key, NULL)) != -ENOENT) {
+  while ((r = rados_nobjects_list_next(list_ctx.get(), &entry, &key, NULL)) != -ENOENT) {
     ASSERT_EQ(0, r);
     rados_ioctx_locator_set_key(ioctx, key);
     ASSERT_EQ(0, rados_remove(ioctx, entry));
   }
-  rados_nobjects_list_close(list_ctx);
+  rados_nobjects_list_close(list_ctx.get());
```

Comment:
I believe Coverity is complaining that the object leaks when we fail the ASSERT_EQ test. See